app/controllers/users_controller.rb

```diff
@@ -1309,6 +1310,38 @@ def read_faq
     render json: success_json
   end
 
+  def recent_searches
+    if !SiteSetting.log_search_queries
+      return render json: failed_json.merge(
+                      error: I18n.t("user_activity.no_log_search_queries")
+                    )
+    end
+
+    query = SearchLog
+      .select(:term)
+      .where(user_id: current_user.id)
+
+    if current_user.user_option.oldest_search_log_date
+      query = query
+        .where("created_at > ?", current_user.user_option.oldest_search_log_date)
+    end
+
+    results = query.group(:term)
+      .order("max(created_at) DESC")
+      .limit(5)
+      .pluck(:term)
+
+    render json: success_json.merge(
+              recent_searches: results
+            )
```

Comment:
```suggestion
    render json: success_json.merge(recent_searches: results)
```

The indentation was a little off so I would just inline it.